### ISSUE TITLE
[Question]LSTM with IMDB sentiment classification task

### ISSUE THREAD

**Issue by xiaoxiamii** [fchollet/keras#1113]
Here I want to use LSTM to do IMDB sentiment classification task, now ,I don't use the embedding layer to generate my word2vector.
I construct the vector by myself.
now , with the problem:
      I use LSTM in the first layer ,so  i need to comfirm the input_shape.
      According to the doc, it tell me the input_shape is (nb_samples, timesteps, input_dim).
     Now, here I show a expamle.
     the longest length of review sentence is 25 ( 19 + 2*3(pad) )
     every word2vector dim is 300
     how can I make the input_shape? 

```
 I make LSTM(128, input_shape=(25,128)) , 
 it export 'Wrong number of dimensions: expected 2, got 3 with shape (50, 25, 300)
 here 50 is the batch_size = 50, the input data struct is [40000,25,300 ]

 then I make LSTM(128, input_shape=(1, 25*128) , 
 data also become [40000,25*300]
 it export 'Wrong number of dimensions: expected 3, got 2 with shape (50, 25*300)
```

 Thanks advance

Here is my simple model struct ：
data.shape is [40000,25,300]

model.Sequential()
model.add( LSTM(128, input_shape=(25, 300)))
# Another struct
###### ###model.add(TimeDistributedDense(label_class))

model.add(Dropout(0.5))

modle.add(Fallten())

model.add(Dense(output_dim = label_class,a ctivation = 'sigmoid'))
model.compile(loss = 'binary_crossentropy', class_mode = 'binary', optimizer='adam')

**Comment by rpinsler** [comment]
Should be something along those lines:
```
nb_samples, timesteps, input_dim = X_train.shape # [40000,25,300]
nb_hidden_nodes = 128
model = Sequential()
model.add(LSTM(nb_hidden_nodes, input_shape=(timesteps, input_dim)))
```
Later on, you call:
```
model.fit(X_train, y_train, batch_size=50)
```
Usually, powers of two are used for batch size though (e.g. 32, 64, 128), but that's just convention I guess.

**Comment by xiaoxiamii** [comment]
@rpinsler I try it  ,but failed. Thank you.
I still find solution to deal with it. 
I update the model struct in first. U can see it.

**Comment by xchen0012** [comment]
I found there are some typos in your another structure. Do you mind updating them? I have same problem like you. Thank you!

**Comment by chinghsienlee** [comment]
Hi all, I am new to Keras. I have questions about example code, 

model = Sequential()
model.add(Embedding(max_features, 128, dropout=0.2))
model.add(LSTM(128, dropout_W=0.2, dropout_U=0.2))  # try using a GRU instead, for fun
model.add(Dense(1))
model.add(Activation('sigmoid'))

1. the LSTM is for each word in a text sentence ?
2. if (1) is yes, what is pooling layer ? whether I need to add a layer between LSTM layer and Dense layer ?
doese the imdb_lstm.py in /examples/ is follow the architecture as below
![image](https://cloud.githubusercontent.com/assets/8901252/18642249/8197072e-7ed2-11e6-9ec8-69b7f36160d2.png) in http://deeplearning.net/tutorial/lstm.html

I am looking for your answers.
Thanks a lot.
